In [1]:
import pandas as pd

url='https://raw.githubusercontent.com/mehakbhartiya/Nike-Sales/refs/heads/main/Data/Nike_Sales_Cleaned.csv'
df = pd.read_csv(url)

df.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 297 entries, 0 to 296
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          297 non-null    int64  
 1   Gender_Category   297 non-null    object 
 2   Product_Line      297 non-null    object 
 3   Product_Name      297 non-null    object 
 4   Size              229 non-null    object 
 5   Units_Sold        297 non-null    float64
 6   MRP               297 non-null    float64
 7   Discount_Applied  297 non-null    float64
 8   Revenue           297 non-null    float64
 9   Order_Date        62 non-null     object 
 10  Sales_Channel     297 non-null    object 
 11  Region            297 non-null    object 
 12  Profit            297 non-null    float64
dtypes: float64(5), int64(1), object(7)
memory usage: 30.3+ KB


In [2]:
print(df['Order_Date'].head(10))
print(df['Order_Date'].dtype)

0    2024-03-09
1    2024-07-09
2           NaN
3    04-10-2024
4    2024/09/12
5           NaN
6    04-06-2025
7           NaN
8    11-10-2024
9           NaN
Name: Order_Date, dtype: object
object


Since the Order_Date type are different we need to clean it

In [3]:
df['Order_Date'] = pd.to_datetime(df['Order_Date'], errors='coerce')

#### After Cleaning

In [4]:
print(df['Order_Date'].head(10))
print(df['Order_Date'].dtype)

0   2024-03-09
1   2024-07-09
2          NaT
3          NaT
4          NaT
5          NaT
6          NaT
7          NaT
8          NaT
9          NaT
Name: Order_Date, dtype: datetime64[ns]
datetime64[ns]


In [24]:
df['Order_Date'].isna().sum()

235

Here the Order_date is converted to datetime and if the order_date cant be converted it is converted to NAT

In [5]:
df[df['Units_Sold'] < 0]

,Order_ID,Gender_Category,Product_Line,Product_Name,Size,Units_Sold,MRP,Discount_Applied,Revenue,Order_Date,Sales_Channel,Region,Profit
8,2008,Women,Soccer,Premier III,11,-1.0,NaN,NaN,0.00,NaT,Online,Hyderabad,-731.18
11,2011,Women,Training,Flex Trainer,L,-1.0,9586.08,NaN,0.00,NaT,Retail,Hyd,1667.08
12,2018,Kids,Lifestyle,Blazer Mid,NaN,-1.0,NaN,0.86,0.00,NaT,Online,Delhi,2109.48
20,2020,Kids,Soccer,Tiempo Legend,L,-1.0,NaN,0.67,0.00,NaT,Retail,Mumbai,439.96
21,2021,Men,Running,Free RN,L,-1.0,7086.70,0.66,-2409.48,NaT,Retail,Hyd,-1120.49
...,...,...,...,...,...,...,...,...,...,...,...,...,...
2410,4410,Men,Basketball,Kyrie Flytrap,XL,-1.0,5967.57,1.10,596.76,2024-01-26,Retail,Mumbai,-1011.68
2457,2019,Men,Lifestyle,Air Force 1,L,-1.0,2452.52,0.12,-2158.22,NaT,Retail,Hyd,-1199.45
2459,4459,Men,Training,SuperRep Go,NaN,-1.0,NaN,0.59,0.00,NaT,Online,Kolkata,3055.56
2460,4460,Women,Soccer,Tiempo Legend,M,-1.0,9682.20,NaN,0.00,NaT,Online,hyderbad,570.42


In the above Query it shows that unit sold can be negative which will give error while procedding for analysis

In [6]:
(df['Units_Sold'] < 0).sum()

205

Here 205 rows contain the negative values which leads to wrong analysis so we need to fix this.

In [8]:
df = df[df['Units_Sold'] >= 0]

#### After Cleaning

In [9]:
(df['Units_Sold'] < 0).sum()

0

Now let’s check if there are any dicounts more than 1. If so, it means it’s referring to a discount above 100%, meaning that the store would pay the customer to take the product.

In [11]:
df[df['Discount_Applied'] > 1]

,Order_ID,Gender_Category,Product_Line,Product_Name,Size,Units_Sold,MRP,Discount_Applied,Revenue,Order_Date,Sales_Channel,Region,Profit
18,2018,Men,Training,Metcon 7,XL,0.0,NaN,1.17,0.00,NaT,Online,bengaluru,-966.89
32,2032,Women,Lifestyle,Blazer Mid,XL,1.0,NaN,1.07,0.00,NaT,Retail,Mumbai,-555.16
37,2037,Kids,Running,Pegasus Turbo,7,3.0,5653.14,1.01,-169.59,NaT,Retail,Pune,2776.96
92,2092,Kids,Basketball,Zoom Freak,XL,2.0,NaN,1.13,0.00,NaT,Online,Delhi,207.20
231,2231,Kids,Training,Flex Trainer,9,0.0,NaN,1.05,0.00,2025-01-04,Online,bengaluru,-39.48
...,...,...,...,...,...,...,...,...,...,...,...,...,...
2371,4371,Men,Lifestyle,Blazer Mid,7,0.0,2042.87,1.25,0.00,NaT,Retail,hyderbad,510.02
2420,4420,Women,Lifestyle,Blazer Mid,L,4.0,4904.79,1.19,-3727.64,2023-11-08,Retail,Pune,2633.82
2443,4443,Men,Lifestyle,Dunk Low,9,3.0,7953.00,1.21,-5010.39,NaT,Retail,Hyd,-665.22
2468,4468,Men,Soccer,Premier III,XL,3.0,6077.13,1.13,-2370.08,2023-11-27,Online,Mumbai,3173.31


In [12]:
(df['Discount_Applied'] > 1).sum()

73

73 rows can similar issue

In [13]:
df = df[df['Discount_Applied'] <= 1]

#### After Cleaning

In [14]:
(df['Discount_Applied'] > 1).sum()

0

Dropping the duplicate ids

In [15]:
df = df.drop_duplicates(subset='Order_ID')

In [16]:
df['Region'].value_counts().head(30)

Region
Kolkata      50
Delhi        50
Mumbai       47
Pune         43
Bangalore    31
bengaluru    27
hyderbad     23
Hyderabad    16
Hyd          10
Name: count, dtype: int64

How many distinct Regions are there in dataset

In [17]:
sorted(df['Region'].dropna().astype(str).str.strip().unique())

['Bangalore',
 'Delhi',
 'Hyd',
 'Hyderabad',
 'Kolkata',
 'Mumbai',
 'Pune',
 'bengaluru',
 'hyderbad']

In [18]:
df['Region'] = (
    df['Region'].astype(str).str.strip().str.lower()
      .replace({'hyd': 'hyderabad', 'hyderbad': 'hyderabad', 'bengaluru': 'bangalore'})
      .str.title()
)

#### After Cleaning

In [19]:
sorted(df['Region'].dropna().astype(str).str.strip().unique())

['Bangalore', 'Delhi', 'Hyderabad', 'Kolkata', 'Mumbai', 'Pune']

In [20]:
df[['MRP','Units_Sold']].isnull().sum()

MRP           157
Units_Sold      0
dtype: int64

In [21]:
mrp_median = df['MRP'].median()
df['MRP'] = df['MRP'].fillna(mrp_median)

In [22]:
df[['MRP','Units_Sold']].isna().sum()

MRP           0
Units_Sold    0
dtype: int64

In [25]:
mrp_median

6170.74

In [26]:
df.to_csv("Nike_Sales_Cleaned.csv", index=False)